ProjectLighthouse.Servers.Website/Controllers/UserPageController.cs

```diff
@@ -87,4 +87,32 @@ public async Task<IActionResult> UnheartUser([FromRoute] int id)
 
         return this.Redirect("~/user/" + id);
     }
+
+    [HttpGet("block")]
+    public async Task<IActionResult> BlockUser([FromRoute] int id)
+    {
+        WebToken? token = this.database.WebTokenFromRequest(this.Request);
+        if (token == null) return this.Redirect("~/login");
+
+        User? blockedUser = await this.database.Users.FirstOrDefaultAsync(u => u.UserId == id);
+        if (blockedUser == null) return this.NotFound();
+
+        await this.database.BlockUser(token.UserId, blockedUser);
+
+        return this.Redirect("~/user/" + id);
+    }
+
+    [HttpGet("unblock")]
+    public async Task<IActionResult> UnblockUser([FromRoute] int id)
+    {
+        WebToken? token = this.database.WebTokenFromRequest(this.Request);
+        if (token == null) return this.Redirect("~/login");
+
+        User? blockedUser = await this.database.Users.FirstOrDefaultAsync(u => u.UserId == id);
+        if (blockedUser == null) return this.NotFound();
+
+        this.database.UnblockUser(token.UserId, blockedUser);
+
+        return this.Redirect("~/user/" + id);
+    }
```

Comment:
You could possibly combine this into one function since they share a lot of code but it might be at the expense of an extra db trip. Might be out of the scope of this PR though.

Reply:
I think considering a lot of our bottlenecks seem to be in the DB I'll save the DB trip